grafana/grafana
# Prometheus: Fix calculating rate interval when there is no interval specified

**What is this feature?**

When there is no specific interval is picked we send the interval value as an empty string. Data source scrape interval will be used under the hood to calculate the rate interval. We recently updated `$__rate_interval` calculation (see it here: https://github.com/grafana/grafana/pull/77234) for an edge case and this case was missed accidentally. This PR is fixing the issue.

**Why do we need this feature?**

Proper `$__rate_interval` calculation

**Who is this feature for?**

All Prometheus users

**Special notes for your reviewer:**

### How to test it?
- Set your Prometheus data source scrape interval as 30s in the config page
- Create a query on a dashboard that is using `$__rate_interval` i.e `sum(rate(process_cpu_seconds_total[$__rate_interval]))`
- Run the query and see it calculates the rate interval as 1m which should be 2m according to the docs https://grafana.com/docs/grafana/latest/datasources/prometheus/template-variables/#use-__rate_interval
- Switch to this branch and refresh the page and see that it is 2m now

Please check that:
